crates/bevy_audio/src/audio_output.rs

```diff
@@ -7,24 +7,26 @@ use std::{collections::VecDeque, io::Cursor};
 
 /// Used to play audio on the current "audio device"
 pub struct AudioOutput {
-    device: Device,
+    device: Option<Device>,
     queue: RwLock<VecDeque<Handle<AudioSource>>>,
 }
 
 impl Default for AudioOutput {
     fn default() -> Self {
         Self {
-            device: rodio::default_output_device().unwrap(),
+            device: rodio::default_output_device(),
             queue: Default::default(),
         }
     }
 }
 
 impl AudioOutput {
     pub fn play_source(&self, audio_source: &AudioSource) {
-        let sink = Sink::new(&self.device);
-        sink.append(Decoder::new(Cursor::new(audio_source.clone())).unwrap());
-        sink.detach();
+        if let Some(device) = &self.device {
```

Comment:
I believe this should be handled at a higher level, currently this would hide the reason audio playback isn't occurring. Possibly instead of using rodio::default_output_device, it could add a resource for a selected audio device from rodio::devices, and use that here instead? Then the app developer could handle which audio device to select (or pass it to the user) and it's not hidden if audio playback is not occurring

Reply:
I thought about that when writing this, I was considering this as a quick fix to the default behavior so that someone just trying out the engine doesn't immediately have a panic. I assumed that it would be changed at some point when the audio engine functionality is more fleshed out. Nonetheless the rodio API is changing completely in the next version, currently in master (see #310), so now I'm not sure about this change as it might be best to just wait for that to drop. @cart any thoughts? I might just close this PR if its not needed.

Reply:
Hmm short term I think its probably best to wait. Medium term once we adapt to the new rodio api, I think we need a way to gracefully handle the following cases:

1. no audio device present (no crashes, playing sounds results in no output)
2. audio device changes (no crashes, as soon as new device is ready audio switches to that device).

Reply:
Maybe we should add something like a "null audio device".

Reply:
That was my thought as well, I'll close this PR for now, that way the initial issue will stay open. #310 has an initial implementation of the new API, could move the conversation over there.

Reply:
awesome. thanks!